mitodl/micromasters
# Changed indexing structure and logic to use a user's program enrollments

#### What are the relevant tickets?

#778 

#### What's this PR do?

Changes our indexing to index one record per user _per program_. A user enrolled in two programs, for example, will now have two records in the index. The enrollment/certificate/etc data for each of those records would only be those that are associated with the appropriate program.

#### Where should the reviewer start?

Probably search/api.py. This might be one that you want to call me over to co-review at some point

#### How should this be manually tested?

Load the realistic users if you haven't already, the run the recreate_index manage command. Once you've done that, you should be able to query ES and see the new indexing format
